### PR TITLE
Use the same Fedora 35 base image for CI and Dev

### PR DESCRIPTION
**Pull request by portante**
To do that, we must remove RPMs which are no longer provided by Fedora 35 and later, `python3-pytest-helpers-namespace` and `python3-tox-current-env`.  The `pytest-helpers-namespace` module is still available from PyPI so will be satisfied by the `pip3` install for the test environment.  The `--current-env` switch is no longer used with `tox` in our testing environment so that package is not needed at all.

The `libpq-devel` RPM dependency fails when building the UBI image when one has multiple PostgreSQL RPM versions.  Rather than trying to solve that problem, it turns out we don't need to have that RPM installed in order leverage the `python3-psycopg2` RPM.

We also drop Fedora 34 from consideration and remove an unused CI related script.

### CODE REVIEW

**Comment by portante** [comment]
The `DO NOT MERGE!!!` label is applied because we must remove the `branch.name` change commit in order for this to work properly on `main`.

**Comment by portante** [comment]
Not sure if this has any impact on PR #3025.

File: jenkins/Makefile
Comment (webbnh) [inline jenkins/Makefile:14]: Trying to keep the lines from wrapping?  😁

File: jenkins/Makefile
Comment (webbnh) [inline jenkins/Makefile:31]: I think these comments are misleading...e.g., these configs are not used for building rpmbuild-containers.

Rather than "all" I think it would be better to enumerate the uses:  "The CI and Fedora development container images are based on Fedora 35" and "The UBI development container image is based on UBI 8.5".  (Although, it would probably be better to avoid including the version number in the comment, since it is clearly specified in the code line; instead just describe the symbols' uses:  "Base image for CI and Fedora development container image" and "Base image for UBI development container image".)

**Review by webbnh — APPROVED** [review]
Looks OK; found some nits, though.

**Review by ndokos — APPROVED** [review]

File: jenkins/branch.name
Comment (dbutenhof) [inline jenkins/branch.name:1]: Just a reminder: likely a good idea to revert this before merging. 😆

**Review by dbutenhof — CHANGES_REQUESTED** [review]

**Review by webbnh — APPROVED** [review]

File: jenkins/Makefile
Comment (dbutenhof) [reply jenkins/Makefile:123]: I think this is arguably wrong, because of the semantics of `.gitignore`. I suspect we're assuming this is run inside `<gitroot>/jenkins` so we're only processing your new `.gitignore` entries and the `jenkins` directory currently isn't deep... however, technically, an unanchored `.gitignore` pattern matches anywhere in the subtree, meaning that `ci.*.Dockerfile` would match `./jenkins/ci.foo.Dockerfile` *and* `./jenkins/something/ci.bar.Dockerfile` so you'd be cleaning just a subset.

This is more equivalent to a `find -exec rm {} \+` although the mechanism of translating a `.gitignore` pattern into a `-name` is a bit more tricky than inclusion.

Like Webb, if for a completely different reason, I guess I'm willing to "hold my nose and move along" because I suspect the *current* local `.gitignore` will work in the current context... but the assumptions here bother me.

**Review by dbutenhof — CHANGES_REQUESTED** [review]
I can't exactly approve until `branch.name` reverts; but while I don't like the new `clean` target I'm otherwise willing to approve.

**Comment by portante** [comment]
So we'll need another review once we have the first round of final approvals where I'll remove the `branch.name` change and push out the new images.

**Review by dbutenhof — DISMISSED** [review]
Without clearing my comment on the branch name, I'll theoretically approve. I'd resolve my other comment, except it's really a reply to *Webb's* comment, so I can't do that. 😆

**Review by webbnh — DISMISSED** [review]
👍

**Review by portante — COMMENTED** [review]
Okay, ready for final review, CI image has been pushed to quay.io.

**Review by dbutenhof — APPROVED** [review]
🧨

**Review by webbnh — APPROVED** [review]
Why am I reviewing this?  It still has the "DO NOT MERGE!!!" label....  😉

**Pull request merged**
